src/v/cluster/scheduling/constraints.cc

```diff
@@ -307,4 +305,30 @@ soft_constraint_evaluator least_disk_filled(
       std::make_unique<impl>(max_disk_usage_ratio, node_disk_reports));
 }
 
+soft_constraint_evaluator
+make_soft_constraint(hard_constraint_evaluator hard_constraint) {
+    class impl : public soft_constraint_evaluator::impl {
+    public:
+        explicit impl(hard_constraint_evaluator hard_constraint)
+          : _hard_constraint(std::move(hard_constraint)) {}
+        uint64_t score(const allocation_node& node) const final {
+            // soft constraint adapts hard constraint to soft one by returning
+            // max score for nodes that matches the soft constraint and 0 for
+            // the ones that not
```

Comment:
nit: this comment deserves to be in the .h file so that the user of the function immediately understands what it does

Reply:
will change